src/cargo-kani/src/call_cbmc_viewer.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use std::ffi::OsString;
+use std::path::Path;
+use std::process::Command;
+
+use crate::session::KaniSession;
+use crate::util::alter_extension;
+
+impl KaniSession {
+    /// Run CBMC appropriately to produce 3 output XML files, then run cbmc-viewer on them to produce a report.
+    pub fn run_visualize(&self, file: &Path) -> Result<()> {
+        let results_filename = alter_extension(file, "results.xml");
+        let coverage_filename = alter_extension(file, "coverage.xml");
+        let property_filename = alter_extension(file, "property.xml");
+
+        {
+            let mut temps = self.temporaries.borrow_mut();
+            temps.push(results_filename.clone());
+            temps.push(coverage_filename.clone());
+            temps.push(property_filename.clone());
+        }
+
+        self.cbmc_variant(file, &["--xml-ui", "--trace"], &results_filename)?;
+        self.cbmc_variant(file, &["--xml-ui", "--cover", "location"], &coverage_filename)?;
+        self.cbmc_variant(file, &["--xml-ui", "--show-properties"], &property_filename)?;
+
+        let args: Vec<OsString> = vec![
+            "--result".into(),
+            results_filename.into(),
+            "--coverage".into(),
+            coverage_filename.into(),
+            "--property".into(),
+            property_filename.into(),
+            "--srcdir".into(),
+            ".".into(), // os.path.realpath(srcdir),
+            "--wkdir".into(),
+            ".".into(), // os.path.realpath(wkdir),
+            "--goto".into(),
+            file.into(),
+            "--reportdir".into(),
+            "report".into(), //reportdir,
+        ];
+
+        // TODO get cbmc-viewer path from self
+        let mut cmd = Command::new("cbmc-viewer");
+        cmd.args(args);
+
+        self.run_suppress(cmd)?;
+
+        Ok(())
+    }
+
+    fn cbmc_variant(&self, file: &Path, extra_args: &[&str], output: &Path) -> Result<()> {
+        let mut args = self.cbmc_flags(file)?;
```

Comment:
Yeah, that's a refactor that needs to happen here that I'd like to try to make a followup PR. I think there's a TODO in `call_cbmc.rs` about it. I'm planning a follow up that will fix the flags we send to cbmc for the 3 runs so that they aren't so much of a mess.